ciao-controller/client.go

```diff
@@ -114,26 +109,48 @@ func (client *ssntpClient) releaseResources(instanceID string) error {
 	return nil
 }
 
-func (client *ssntpClient) removeInstance(instanceID string) {
-	err := client.releaseResources(instanceID)
+func (client *ssntpClient) removeInstance(i *types.Instance) {
```

Comment:
possibly I was also trying to avoid multiple calls to GetInstance in the other routines.